analyzers/tests/SonarAnalyzer.UnitTest/AnalysisContext/SonarAnalysisContextTest.Register.cs

```diff
@@ -132,6 +131,86 @@ public void RegisterCodeBlockStartAction_UnchangedFiles_SonarAnalysisContext(str
         context.AssertDelegateInvoked(expected);
     }
 
+    [TestMethod]
+    public void SonarCompilationStartAnalysisContext_RegisterCompilationEndAction()
+    {
+        var context = new DummyAnalysisContext(TestContext);
+        var startContext = new DummyCompilationStartAnalysisContext(context);
+        var sut = new SonarCompilationStartAnalysisContext(new(context, DummyMainDescriptor), startContext);
+        sut.RegisterCompilationEndAction(_ => { });
+
+        startContext.AssertExpectedInvocationCounts(expectedCompilationEndCount: 1);
+    }
+
+    [TestMethod]
+    public void SonarCompilationStartAnalysisContext_RegisterSemanticModel()
+    {
+        var context = new DummyAnalysisContext(TestContext);
+        var startContext = new DummyCompilationStartAnalysisContext(context);
+        var sut = new SonarCompilationStartAnalysisContext(new(context, DummyMainDescriptor), startContext);
+        sut.RegisterSemanticModelAction(_ => { });
+
+        startContext.AssertExpectedInvocationCounts(expectedSemanticModelCount: 1);
+    }
+
+    [TestMethod]
+    public void SonarCompilationStartAnalysisContext_RegisterSymbolAction()
+    {
+        var context = new DummyAnalysisContext(TestContext);
+        var startContext = new DummyCompilationStartAnalysisContext(context);
+        var sut = new SonarCompilationStartAnalysisContext(new(context, DummyMainDescriptor), startContext);
+        sut.RegisterSymbolAction(_ => { });
+
+        startContext.AssertExpectedInvocationCounts(expectedSymbolCount: 1);
```

Comment:
I usually complain about these named arguments, but in this case, it makes perfect sense.